### ISSUE TITLE
On wimpsim_reader (Trac #2156)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2148]
Found when running this script:

```text
from I3Tray import *
from icecube import icetray, dataclasses, dataio
from icecube import simclasses, wimpsim_reader, phys_services
from icecube.wimpsim_reader import WimpSimReaderEarth

tray = I3Tray()

tray.AddService("I3SPRNGRandomServiceFactory","Random",
                NStreams = 2,
                Seed = 42,
                StreamNum = 1,
                InstallServiceAs = "I3RandomService")

tray.AddSegment(WimpSimReaderEarth,"EarthWimpsim-reader",
                Infile = '/data/user/grenzi/data/EarthWimpData/Earth/we-m50-ch11-earth.010010.000100.dat',
                #GCDFileName = '/data/user/grenzi/data/EarthWimpData/gcd/GeoCalibDetectorStatus_IC86.55697_V2.i3.gz'
                StartMJD=55555,
                EndMJD=55666,
               )

def prettyprint(frame):
  icetray.logging.log_info("=====================")
  icetray.logging.log_info(str(frame["I3EventHeader"]))
  icetray.logging.log_info(str(frame["WIMP_params"]))
  icetray.logging.log_info(str(frame["I3MCTree"]))
tray.AddModule(prettyprint, "print",
  Streams = [icetray.I3Frame.Physics, icetray.I3Frame.DAQ])

tray.AddModule( 'I3Writer', 'EventWriter',
                Filename          = "/data/user/grenzi/data/EarthWimpData/Earth/we-m50-ch11-earth.010010.000100.i3.bz2",
                #Streams           = [icetray.I3Frame.Physics, icetray.I3Frame.DAQ],
                DropOrphanStreams = [icetray.I3Frame.DAQ]
                )

tray.Execute()

del tray

```

The segments in this submodule: http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/wimpsim-reader/trunk/python/wimpsimreader.py

Are set with not acceptable values of ```InjectionRadius```

Both:

```text
29	    InjectionRadius = 0*I3Units.meter , #default 0*I3Units.meter
```

```text
74	    InjectionRadius = 0*I3Units.meter , #default 0*I3Units.meter
```

While in /IceCube/projects/wimpsim-reader/trunk/private/wimpsim-reader/I3WimpSimReader.cxx we have:

```text
180	  if (radius_<=0.)
181	    log_fatal("Injection radius must be positive and not zero");
```

Hence, running this segments with whatever the inputs gives the error:

```text
RuntimeError: Injection radius must be positive and not zero (in virtual void I3WimpSimReader::Configure())
```

Also, it is commented that default is 0, but the log message when running is 

```text
InjectionRadius
    Description :  If >0, events will be injected in cylinder with zmin, zmax height
    Default     :  nan
    Configured  :  0.0
``` 

Migrated from https://code.icecube.wisc.edu/ticket/2156
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:23", 
    "description": "Found when running this script:\n\n{{{\nfrom I3Tray import *\nfrom icecube import icetray, dataclasses, dataio\nfrom icecube import simclasses, wimpsim_reader, phys_services\nfrom icecube.wimpsim_reader import WimpSimReaderEarth\n\ntray = I3Tray()\n\ntray.AddService(\"I3SPRNGRandomServiceFactory\",\"Random\",\n                NStreams = 2,\n                Seed = 42,\n                StreamNum = 1,\n                InstallServiceAs = \"I3RandomService\")\n\ntray.AddSegment(WimpSimReaderEarth,\"EarthWimpsim-reader\",\n                Infile = '/data/user/grenzi/data/EarthWimpData/Earth/we-m50-ch11-earth.010010.000100.dat',\n                #GCDFileName = '/data/user/grenzi/data/EarthWimpData/gcd/GeoCalibDetectorStatus_IC86.55697_V2.i3.gz'\n                StartMJD=55555,\n                EndMJD=55666,\n               )\n\ndef prettyprint(frame):\n  icetray.logging.log_info(\"=====================\")\n  icetray.logging.log_info(str(frame[\"I3EventHeader\"]))\n  icetray.logging.log_info(str(frame[\"WIMP_params\"]))\n  icetray.logging.log_info(str(frame[\"I3MCTree\"]))\ntray.AddModule(prettyprint, \"print\",\n  Streams = [icetray.I3Frame.Physics, icetray.I3Frame.DAQ])\n\ntray.AddModule( 'I3Writer', 'EventWriter',\n                Filename          = \"/data/user/grenzi/data/EarthWimpData/Earth/we-m50-ch11-earth.010010.000100.i3.bz2\",\n                #Streams           = [icetray.I3Frame.Physics, icetray.I3Frame.DAQ],\n                DropOrphanStreams = [icetray.I3Frame.DAQ]\n                )\n\ntray.Execute()\n\ndel tray\n\n}}}\n\nThe segments in this submodule: http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/wimpsim-reader/trunk/python/wimpsimreader.py\n\nAre set with not acceptable values of {{{InjectionRadius}}}\n\nBoth:\n\n{{{\n29\t    InjectionRadius = 0*I3Units.meter , #default 0*I3Units.meter\n}}}\n\n{{{\n74\t    InjectionRadius = 0*I3Units.meter , #default 0*I3Units.meter\n}}}\n\nWhile in /IceCube/projects/wimpsim-reader/trunk/private/wimpsim-reader/I3WimpSimReader.cxx we have:\n\n{{{\n180\t  if (radius_<=0.)\n181\t    log_fatal(\"Injection radius must be positive and not zero\");\n}}}\n\nHence, running this segments with whatever the inputs gives the error:\n\n{{{\nRuntimeError: Injection radius must be positive and not zero (in virtual void I3WimpSimReader::Configure())\n}}}\n\nAlso, it is commented that default is 0, but the log message when running is \n{{{\nInjectionRadius\n    Description :  If >0, events will be injected in cylinder with zmin, zmax height\n    Default     :  nan\n    Configured  :  0.0\n}}} \n\n", 
    "reporter": "grenzi", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067323910946", 
    "component": "combo simulation", 
    "summary": "On wimpsim_reader", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-05-17T15:55:29", 
    "milestone": "", 
    "owner": "mjl5147", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180922T09:30:50: 
* olivas commented:

> wimpsim-reader needs an owner.


* olivas changed owner from "" to "olivas"
* olivas changed milestone from "" to "Winter Solstice 2018"
* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20181114T14:45:42: olivas changed owner from "olivas" to ""

**Comment by icecube-trac** [comment]
Trac update at 20181204T15:01:06: 
* mjl5147 changed status from "assigned" to "accepted"
* mjl5147 changed owner from "" to "mjl5147"

**Comment by icecube-trac** [comment]
Trac update at 20181204T16:55:41: 
* mjl5147 changed status from "accepted" to "closed"
* mjl5147 changed resolution from "" to "fixed"
* mjl5147 commented:

> In changeset 167286:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="167286"
> Clean up some code in I3WimpSimReader related to the injection of events. The code assumed that NAN would be used for the InjectionRadius if, for some reason, you wanted to inject in a box (all of the tests, segments, and examples assume you'd use a cylinder). It now accepts 0 as well as NAN. I also added a bit of text to the config options to specify that the PositionLimits is still used to calculate the center of the cylinder. Passes all 2 included tests and the example script runs to completion. Closes #2148.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190112T04:09:32: nwhitehorn commented:

> In changeset 170618:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170618"
> Clean up some code in I3WimpSimReader related to the injection of events. The code assumed that NAN would be used for the InjectionRadius if, for some reason, you wanted to inject in a box (all of the tests, segments, and examples assume you'd use a cylinder). It now accepts 0 as well as NAN. I also added a bit of text to the config options to specify that the PositionLimits is still used to calculate the center of the cylinder. Passes all 2 included tests and the example script runs to completion. Closes #2148.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:23: 
* olivas commented:

> Milestone Winter Solstice 2018 deleted


* olivas changed milestone from "Winter Solstice 2018" to ""
